python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -986,7 +986,13 @@ def placement_groups_to_resource_demands(
     resource_demand_vector = []
     unconverted = []
     for placement_group in pending_placement_groups:
-        shapes = [dict(bundle.unit_resources) for bundle in placement_group.bundles]
+        # Skip **placed** bundle (which has node id associated with it).
```

Comment:
Is this behavior correct with `STRICT_PACK`? If already placed bundles are removed, will the new bundles be placed on different nodes?

Reply:
Yes, the behavior is still correct with `STRICT_PACK` because we only calculate shapes here, and shapes here do not directly instruct scheduling. The scheduler will not schedule a `STRICT_PACK` group across different nodes.

Back to the duty of this `placement_groups_to_resource_demands` function, it is correct to present the remaining demand of a `STRICT_PACK` group to the later `get_bin_pack_residual`. Say we have 3 bundles in a `STRICT_PACK` but 2 of the bundles have node_id on them, we still need to present the remaining 1 bundle to later `get_bin_pack_residual` and then `get_bin_pack_residual` should consume it.